// csx/rr/Complex useState,useEffect hook assignment/product-dashboard/src/App.jsx
import React, { useState } from "react";

function App() {
  const products = [
    { id: 1, name: "Laptop", price: 80000, category: "Premium" },
    { id: 2, name: "Mouse", price: 500, category: "Basic" },
    { id: 3, name: "Keyboard", price: 1500, category: "Basic" },
    { id: 4, name: "Smartphone", price: 60000, category: "Premium" },
    { id: 5, name: "Monitor", price: 12000, category: "Deluxe" },
    { id: 6, name: "Headphones", price: 3000, category: "Deluxe" }
  ];

  const [category, setCategory] = useState("All");

 
  const filteredProducts =
    category === "All"
      ? products
      : products.filter((p) => p.category === category);

  return (
    <div style={{ padding: "20px" }}>
      <h2>Product Dashboard</h2>

    
      <select
        value={category}
        onChange={(e) => setCategory(e.target.value)}
        style={{ padding: "8px", marginBottom: "20px" }}
      >
        <option value="All">All</option>
        <option value="Premium">Premium</option>
        <option value="Deluxe">Deluxe</option>
        <option value="Basic">Basic</option>
      </select>

     
      <div
        style={{
          display: "grid",
          gridTemplateColumns: "repeat(auto-fill, minmax(200px, 1fr))",
          gap: "15px"
        }}
      >
        {filteredProducts.map((product) => (
          <div
            key={product.id}
            style={{
              border: "1px solid #ddd",
              borderRadius: "10px",
              padding: "15px",
              boxShadow: "0 2px 6px rgba(0,0,0,0.1)",
              backgroundColor: "#fff"
            }}
          >
            <h3>{product.name}</h3>
            <p>₹{product.price}</p>

            {/* Category Badge */}
            <span
              style={{
                padding: "5px 10px",
                borderRadius: "20px",
                fontSize: "12px",
                backgroundColor:
                  product.category === "Premium"
                    ? "#ffd700"
                    : product.category === "Deluxe"
                    ? "#87cefa"
                    : "#90ee90"
              }}
            >
              {product.category}
            </span>
          </div>
        ))}
      </div>
    </div>
  );
}

export default App;
