// csx/class-question-assignment/image-slider.jsx
import React from 'react'
import { useState } from 'react'


const images = [
 "https://images.unsplash.com/photo-1506744038136-46273834b3fb",
 "https://images.unsplash.com/photo-1500530855697-b586d89ba3ee",
 "https://images.unsplash.com/photo-1493246507139-91e8fad9978e",
 "https://images.unsplash.com/photo-1519125323398-675f0ddb6308"
];
function App() {
const [currIndex, setcurrIndex] = useState(0);

 const handlePrev = () => {
  if (currIndex == 0)
   setcurrIndex(images.length - 1);
  else
   setcurrIndex(currIndex - 1);
 }

 const handleNext = () => {
  if (currIndex == images.length - 1)
   setcurrIndex(0);
  else
   setcurrIndex(currIndex + 1);
 }

 return (
  <>
   <h1>Image Slider</h1>
   <div
    style={{
     display: "flex",
     flexDirection: "column",
     alignItems: "center"
    }}>
    <p>{currIndex + 1}</p>

    <img
     src={images[currIndex]}
     height="400"
     width="600"
     style={{
      borderRadius: "12px",
      objectFit: "cover",
      boxShadow: "0px 4px 10px rgba(0,0,0,0.3)",
      marginTop: "10px"
     }}
    />

   </div>
   <button onClick={(e) => handlePrev(e)}>Previous</button>
   <button onClick={(e) => handleNext(e)}>Next</button>
  </>
 )
}

export default App
