// csx/local_storage/form+local_storage+card.jsx
import React, { useState, useEffect } from "react";

function App() {


  const [name, setName] = useState("");
  const [course, setCourse] = useState("");
  const [students, setStudents] = useState([]);

  useEffect(() => {
    const savedData = localStorage.getItem("students");

    if (savedData) {
      setStudents(JSON.parse(savedData));
    }
  }, []);


  useEffect(() => {

    localStorage.setItem(
      "students",
      JSON.stringify(students)
    );
  }, [students]);


  const addStudent = () => {

    if (name === "" || course === "") {
      alert("Please fill all fields");
      return;
    }

    const newStudent = {
      id: Date.now(),
      name: name,
      course: course
    };

    setStudents([...students, newStudent]);

    setName("");
    setCourse("");

  };

  return (
    <div>

      <h1>Student Form</h1>

      <input
        type="text"
        placeholder="Enter Name"
        value={name}
        onChange={(e) => setName(e.target.value)}
      />

      <br />
      <br />

      <input
        type="text"
        placeholder="Enter Course"
        value={course}
        onChange={(e) => setCourse(e.target.value)}
      />

      <br />
      <br />

      <button onClick={addStudent}>
        Add Student
      </button>

      <hr />

      <h2>Student Cards</h2>

      {

        students.map((student) => (

          <div
            key={student.id}
            style={{
              border: "1px solid black",
              padding: "10px",
              margin: "10px",
              width: "250px"
            }}
          >

            <h3>{student.name}</h3>

            <p>
              Course: {student.course}
            </p>

          </div>

        ))

      }

    </div>
  );
}

export default App;
